samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -606,118 +610,65 @@ static struct bt_bap_stream_ops stream_ops = {
 };
 
 #if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
-struct find_valid_bis_data {
+struct bis_channel_allocation_data {
 	struct {
+		bool chan_allocation_available;
 		uint8_t index;
 		enum bt_audio_location chan_allocation;
 	} bis[BT_ISO_BIS_INDEX_MAX];
 
 	uint8_t cnt;
+	struct bt_bap_base_codec_id bap_base_codec_id;
 };
 
 /**
  * This is called for each BIS in a subgroup
  *
- * It returns `false` if the current BIS contains all of the channels we are looking for,
- * or if it does not contain any and we are looking for BT_AUDIO_LOCATION_MONO_AUDIO. This stops
- * the iteration of the remaining BIS in the subgroup.
- *
- * It returns `true` if the BIS either contains none or some of the channels we are looking for.
- * If it contains some, then that is being stored in the user_data, so that the calling function
- * can check if a combination of the BIS satisfy the channel allocations we want.
+ * Gets BIS channel allocation (if exists).
+ * Always returns `true` to continue to next BIS
  */
-static bool find_valid_bis_cb(const struct bt_bap_base_subgroup_bis *bis,
-					       void *user_data)
+static bool bis_get_channel_allocation_cb(const struct bt_bap_base_subgroup_bis *bis,
+					  void *user_data)
 {
-	struct find_valid_bis_data *data = user_data;
-	struct bt_audio_codec_cfg codec_cfg = {0};
+	struct bis_channel_allocation_data *data = user_data;
+	struct bt_audio_codec_cfg codec_cfg;
 	enum bt_audio_location chan_allocation;
-	int err;
+	int err, idx;
+
+	idx = data->cnt++;
+	data->bis[idx].index = bis->index;
+	data->bis[idx].chan_allocation_available = false;
 
 	err = bt_bap_base_subgroup_bis_codec_to_codec_cfg(bis, &codec_cfg);
 	if (err != 0) {
 		printk("Could not get codec configuration for BIS: %d\n", err);
-		return true;
+
+		return true; /* continue to next BIS */
 	}
 
+	/* bt_bap_base_subgroup_bis_codec_to_codec_cfg only updates the codec specific capabilities
+	 * data. The bap_base_codec_id data in the codec_cfg struct needed by
+	 * bt_audio_codec_cfg_get_chan_allocation for the fallback check.
+	 */
```

Comment:
I will add clarification in the comment

Reply:
IMO it would be better to check if the codec ID is LC3 in the BASE in `subgroup_get_valid_bis_indexes_cb` and then stop parsing the BASE if it is not. LC3 is the only codec we support, so parsing non-LC3 subgroups is moot. 

If we apply this check, then it it is always LC3 when we reach this point, and we can just hardcode the `data.bap_base_codec_id.id = BT_HCI_CODING_FORMAT_LC3`

Reply:
Okay. I will do that :-)